lib/numorstring/uint8orstring.go

```diff
@@ -54,10 +54,10 @@ func (i *Uint8OrString) UnmarshalJSON(b []byte) error {
 
 // MarshalJSON implements the json.Marshaller interface.
 func (i Uint8OrString) MarshalJSON() ([]byte, error) {
-	if num, err := i.NumValue(); err == nil {
-		return json.Marshal(num)
-	} else {
+	if i.Type == NumOrStringString {
```

Comment:
Will do - I forgot that I made this change, it was a loooong time ago.  Can't recall immediately why but it was causing me some trouble.